### DISCUSSION TITLE
p-calendar: Revamp to support of selection of date ranges, allow open ended ranges

### DISCUSSION THREAD

**Discussion by axos88** [primefaces/community#805] [PrimeNG] [answered]
I'd like to request a feature to allow selection of open ended date ranges with the p-calendar copmponent.
There are a few changes that I imagine would be necessary to make this ergonomic:

1. Two buttons in the buttonBar to allow selection of "From beginning of time" and "Until the end of time".
2. The selected dates/date ranges need to be exposed in an event (currently only the selectedDate is available, which gets called on each click). Open ends could be represented by nulls. 
3. It would be great if the component would allow saving the range start and range end into different responsive form controls.
4. Allow selecting either end of the range first, and it should correctly reverse the ordering in case the end was selected first
5. +1: When selecting a date range, I'd expect to have the ability to highlight the dates between the currently hovered date and the already selected end.
6. +2: Add ability to customize the formatting of the displayed text in the input box, probably via some callback function

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

Right now we're fully focused on passthrough implementation and unstyled mode. We do not have a plan to implement such a feature to the Calendar component. We've added your request to our internal backlog. 

Thanks a lot for the feedback!
